src/linux_mcp_server/connection/ssh.py

```diff
@@ -291,6 +292,22 @@ async def execute_command(
     return await _execute_local(command)
 
 
+async def get_bin_path(command: str, hostname: Host, required: bool = False) -> str | None:
+    """Get the full path to an executable on a remote system.
+
+    Raises VauleError if not found and required is True.
+    """
+    bin_path = None
+    rc, out, err = await execute_command(["command", "-v", command], hostname)
+    if rc == 0:
+        return out.strip()
+
+    if required:
+        raise ValueError(f"Unable to find '{command}' on {hostname}")
```

Comment:
Hmm, to me, it feels like if this check should belong to the tool, as we can return a `ToolError()`  (even though, it's not a tool error per se, but a missing binary error)

I see that you are using this same function in both modes (required=true, required=false) in different tools, and in both, you have a duplication of this check (kinda). 

This feels a bit confusing to me, so, either;

* let's remove the `required` parameter from this function, and handle it at the tool level
* let's rely on the error being raised here, avoiding checking on multiple places the same thing